src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -726,10 +727,25 @@ def get_store_info(self) -> ServerModel:
 
         Returns:
             Information about the store.
+
+        Raises:
+            KeyError: If the deployment ID could not be loaded from the
+                database.
         """
         model = super().get_store_info()
         sql_url = make_url(self.config.url)
         model.database_type = ServerDatabaseType(sql_url.drivername)
+
+        # Fetch the deployment ID from the database and use it to replace the one
+        # fetched from the global configuration
+        with Session(self.engine) as session:
+            identity = session.exec(select(IdentitySchema)).first()
+
+            if identity is None:
```

Comment:
Are you sure this should be an exception? Wouldnt it suck to have this happen? What would be a way to recover

Reply:
I guess you could insert `str(GlobalConfiguration().user_id).replace("-", "")` again. However, I am not recommending this since it might hide unintended bugs that screw with our analytics. Overall, `identity is None` should never happen unless the user intentionally screwed with the DB using SQL which we can never fully recover from.

Reply:
Yes, I had my doubts about making this an exception, but the I realized that if this row is missing from the database, something terribly wrong must've happened during the database migration, which we absolutely want to catch early.

Reply:
in this case, i would tell the user to create a new database or do something. this exception would be incredibly difficult to deal with otherwise